src/v/raft/consensus.cc

```diff
@@ -3488,6 +3488,10 @@ void consensus::maybe_update_last_visible_index(model::offset offset) {
     _visibility_upper_bound_index = std::max(
       _visibility_upper_bound_index, offset);
     _majority_replicated_index = std::max(_majority_replicated_index, offset);
+    if (is_elected_leader()) {
```

Comment:
Do you mean this function or maybe_update_majority_replicated_index? Because maybe_update_last_visible_index is called for the followers as well I think.

Reply:
i meant the one called on the leader, sorry

Reply:
right, I can replace a conditional with a clarifying comment

Reply:
great, other than that it all looks great